gui/default/syncthing/core/syncthingController.js

```diff
@@ -3077,6 +3077,95 @@ angular.module('syncthing.core')
                     address.indexOf('unix://') == 0 ||
                     address.indexOf('unixs://') == 0);
         };
+
+        $scope.copyToClipboard = function (event, text) {
+            console.log(text);
+            var success = $translate.instant("Copied to clipboard!");
+            var failure = $translate.instant("Copy to clipboard failed!");
+
+            if (navigator.clipboard && navigator.clipboard.writeText) {
+                // Default for modern browsers on localhost or HTTPS.
+                navigator.clipboard.writeText(text);
+                return alert(success);
```

Comment:
I've changed the code following your second suggestion. However, do you think it's safe to default to `success` like that rather than default to `failure` and change the message to `success` only in the successful cases instead?

Reply:
As long as this function is short enough to view all at once, yes it's safe.  It's easy enough to verify that each case either sets the failure message or is actually successful.  The suggested default was just considering which way needs less lines of code.